template.rb

```diff
@@ -92,23 +92,33 @@ def copy_solidus_starter_frontend_files
   copy_file 'config/initializers/devise.rb', 'tmp/devise.rb'
   append_file 'config/initializers/devise.rb', File.read('tmp/devise.rb')
 
-  application <<~RUBY
-    if Rails.autoloaders.main
-      # Rails 7
+  if Rails.gem_version >= Gem::Version.new('7')
+    application <<~RUBY
       Rails.autoloaders.main.ignore(Rails.root.join('app/monkey_patches'))
-    else
-      # Rails 6 and older
+
+      config.to_prepare do
+        Dir.glob(Rails.root.join('app/monkey_patches/**/*_monkey_patch.rb')).each do |monkey_patch|
+          load monkey_patch
+        end
+      end
```

Comment:
I duplicated the `to_prepare` block in each branch of the conditional to ensure that it gets written together with the `Rails.autoloaders.main.ignore` block.